ui/app/components/auth-jwt.js

```diff
@@ -16,129 +17,149 @@ const ERROR_MISSING_PARAMS =
-  },
+  }
 
   closeWindow(oidcWindow) {
     this.watchPopup.cancelAll();
     this.watchCurrent.cancelAll();
     oidcWindow.close();
-  },
+  }
 
   handleOIDCError(err) {
-    this.onLoading(false);
     this.prepareForOIDC.cancelAll();
-    this.onError(err);
-  },
+    this.args.onError(err);
+  }
 
   // NOTE TO DEVS: Be careful when updating the OIDC flow and ensure the updates
   // work with implicit flow. See issue https://github.com/hashicorp/vault-plugin-auth-jwt/pull/192
-  prepareForOIDC: task(function* (oidcWindow) {
-    const thisWindow = this.getWindow();
-    // show the loading animation in the parent
-    this.onLoading(true);
+  prepareForOIDC = task(async (oidcWindow) => {
+    const thisWindow = this.getWindow;
+
     // start watching the popup window and the current one
     this.watchPopup.perform(oidcWindow);
     this.watchCurrent.perform(oidcWindow);
     // wait for message posted from oidc callback
     // see issue https://github.com/hashicorp/vault/issues/12436
     // ensure that postMessage event is from expected source
+    // eslint-disable-next-line no-constant-condition
     while (true) {
-      const event = yield waitForEvent(thisWindow, 'message');
+      const event = await waitForEvent(thisWindow, 'message');
       if (event.origin === thisWindow.origin && event.isTrusted && event.data.source === 'oidc-callback') {
         return this.exchangeOIDC.perform(event.data, oidcWindow);
       }
-      // continue to wait for the correct message
     }
-  }),
+  });
 
-  watchPopup: task(function* (oidcWindow) {
+  watchPopup = task(async (oidcWindow) => {
+    // eslint-disable-next-line no-constant-condition
     while (true) {
       const WAIT_TIME = Ember.testing ? 50 : 500;
 
-      yield timeout(WAIT_TIME);
+      await timeout(WAIT_TIME);
       if (!oidcWindow || oidcWindow.closed) {
         return this.handleOIDCError(ERROR_WINDOW_CLOSED);
       }
     }
-  }),
+  });
 
-  watchCurrent: task(function* (oidcWindow) {
+  watchCurrent = task(async (oidcWindow) => {
     // when user is about to change pages, close the popup window
-    yield waitForEvent(this.getWindow(), 'beforeunload');
+    await waitForEvent(this.getWindow, 'beforeunload');
     oidcWindow.close();
-  }),
+  });
 
-  exchangeOIDC: task(function* (oidcState, oidcWindow) {
+  exchangeOIDC = task(async (oidcState, oidcWindow) => {
     if (oidcState === null || oidcState === undefined) {
       return;
     }
-    this.onLoading(true);
 
     let { namespace, path, state, code } = oidcState;
 
     // The namespace can be either be passed as a query parameter, or be embedded
     // in the state param in the format `<state_id>,ns=<namespace>`. So if
     // `namespace` is empty, check for namespace in state as well.
-    if (namespace === '' || this.flagsService.hvdManagedNamespaceRoot) {
+    // TODO smoke test HVD flag here and add test
+    if (namespace === '' || this.flags.hvdManagedNamespaceRoot) {
```

Comment:
the check for HVD was added by this PR: https://github.com/hashicorp/vault/pull/11283 but there's no test or deeper explanation as to why this fixed the bug (the changelog was copy pasta'd 🍝 ) 

I'll investigate and add a test in a follow-on PR